src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/VsProjectAdapter.cs

```diff
@@ -365,27 +364,56 @@ public async Task<NuGetFramework> GetTargetFrameworkAsync()
 
         public async Task<string> GetRestorePackagesWithLockFileAsync()
         {
-            await _threadingService.JoinableTaskFactory.SwitchToMainThreadAsync();
+            return await GetPropertyValueAsync(ProjectBuildProperties.RestorePackagesWithLockFile);
+        }
+
+        public async Task<string> GetNuGetLockFilePathAsync()
```

Comment:
```suggestion
        public Task<string> GetNuGetLockFilePathAsync()
        {
            return GetPropertyValueAsync(ProjectBuildProperties.NuGetLockFilePath);
        }
```

Saves some allocations.

Reply:
Updated